src/matlab2tikz.m

```diff
@@ -998,7 +998,9 @@ function addCustomCode(fid, before, code, after)
     backgroundColor = get(handle, 'Color');
     if ~isNone(backgroundColor)
         [m2t, col] = getColor(m2t, handle, backgroundColor, 'patch');
-        if ~strcmp(col, 'white')
+        % Fill background if not just white or if multiple axes and current
+        % is visible
+        if ~strcmp(col, 'white') || (numel(m2t.relevantAxesHandles) > 1 && isVisible(handle))
```

Comment:
Yes, many subplots will have white bg, as does the Matlab figure. However, physically printing it on colored paper will not print white bg...

I tested the whole ACID and I can put before after .pdfs if necessary. 

The overlapping test is something I thought about but then it is really not worth the effort for now, since things work fine (I had to ensure axis is visible and without that check, ACID(14) was breaking).